Set variant media correctly when added to cart

willopez opened the pull request:
Resolves: #5062
Impact: **Critical**
Type: **bugfix**

## Issue
Incorrect variant image rendered in mini cart popover and cart page

## Solution
Give priority to media in variant over media stored on the parent product object. 

## Testing
1. Add variant other than the first one to the shopping cart
2. Confirm the correct variant image is rendered in the cart popover
3. Confirm the correct variant images are rendered in the cart page

aldeed commented on imports/plugins/core/graphql/server/no-meteor/xforms/cart.js:
```diff
@@ -56,8 +56,17 @@ async function xformCartItem(context, catalogItems, products, cartItem) {
 
+  const variantSourceProduct = products.find((product) => product._id === variantId);
+  const catalogVariantId = variantSourceProduct.ancestors[1];
+  const catalogVariant = catalogProduct.variants.find((cVariant) => cVariant._id === catalogVariantId);
```
`ancestors[1]` is not correct because sometimes there's only one item in the `ancestors` array. `variantId` here may be referring to a variant or option ID, and we've already found `variant` (which may be an option) with the `findVariantInCatalogProduct` call above.

So I think removing everything above here and changing `catalogVariant` to `variant` below will make this correct.

This would be a good time to add a unit test to prove this is working, too.

aldeed commented:
Also sign your commits

willopez commented:
@aldeed I have addressed your comments. In the interest of time, I would like to get this merged. I have created another issue for myself to add a unit test in another PR. https://github.com/reactioncommerce/reaction/issues/5086
